packages/genesys-spark-components/src/components/beta/gux-avatar-group/gux-avatar-group.tsx

```diff
@@ -0,0 +1,223 @@
+import { Component, h, JSX, Element, State, Prop, Listen } from '@stencil/core';
+import { trackComponent } from '@utils/tracking/usage';
+import { GuxAvatarAccent, GuxAvatarSize } from '../gux-avatar/gux-avatar.types';
+import { buildI18nForComponent, GetI18nValue } from '../../../i18n';
+import { logWarn } from '@utils/error/log-error';
+
+import {
+  first,
+  last,
+  next,
+  previous
+} from 'components/stable/gux-list/gux-list.service';
+import defaultResources from './i18n/en.json';
+import { GuxAvatarGroupLimit } from './gux-avatar-group.types';
+
+const validFocusableItems = [
+  'gux-avatar-list-item-beta',
+  'gux-avatar-overflow-beta'
+];
+
+interface ProcessedAvatar {
+  avatar: HTMLGuxAvatarBetaElement;
+  interactiveElement?: HTMLElement;
+  img?: HTMLImageElement | null;
+  name: string;
+  accent: GuxAvatarAccent;
+  isOverflow: boolean;
+}
+
+@Component({
+  styleUrl: 'gux-avatar-group.scss',
+  tag: 'gux-avatar-group-beta',
+  shadow: true
+})
+export class GuxAvatarGroup {
+  private i18n: GetI18nValue;
+  private groupRef: HTMLElement;
+
+  @Element()
+  root: HTMLElement;
+
+  @Prop()
+  avatarLimit: GuxAvatarGroupLimit = 7;
+
+  @State()
+  private processedAvatars: ProcessedAvatar[] = [];
+
+  async componentWillLoad(): Promise<void> {
+    this.processAvatars();
+    this.i18n = await buildI18nForComponent(this.root, defaultResources);
+    trackComponent(this.root);
+  }
+
+  componentWillRender(): void {
+    this.processAvatars();
+  }
+
+  private getAccent(index): string {
+    return `${(index % 12) + 1}` as GuxAvatarAccent;
+  }
+
+  @Listen('keydown')
+  onKeyDown(event: KeyboardEvent): void {
+    switch (event.key) {
+      case 'ArrowLeft':
+        event.preventDefault();
+        previous(this.groupRef, validFocusableItems);
+        break;
+      case 'ArrowRight':
+        event.preventDefault();
+        next(this.groupRef, validFocusableItems);
+        break;
+      case 'Home':
+        event.preventDefault();
+        first(this.groupRef, validFocusableItems);
+        break;
+      case 'End':
+        event.preventDefault();
+        last(this.groupRef, validFocusableItems);
+        break;
+    }
+  }
+
+  private validateChildElements(childElements: HTMLElement[]) {
+    if (childElements.length === 0) {
+      return;
+    }
+
+    // Check if avatarElements are buttons, anchor tags or gux-avatar beta and log warning if there are invalid elements
+    const validTagNames = ['GUX-AVATAR-BETA', 'BUTTON', 'A'];
+    const invalidElements = childElements.some(
+      el => !validTagNames.includes(el.tagName)
+    );
+
+    if (invalidElements) {
+      logWarn(
+        this.root,
+        'gux-avatar-group-beta: Invalid child element detected. All child elements must be either buttons, anchor tags or gux-avatar-beta components'
+      );
+    }
+  }
+
+  private processAvatars(): void {
+    const avatarElements = Array.from(this.root.children) as HTMLElement[];
```

Comment:
I've created an alternate version of this which uses an avatarData prop set with JS instead of trying to work with the children of the avatar-group. `[feature/COMUI-3049_data-prop](https://github.com/MyPureCloud/genesys-spark/tree/feature/COMUI-3049_data-prop)`.